pyam/utils.py

```diff
@@ -329,21 +329,28 @@ def convert_r_columns(c):
     df.loc[df.unit.isnull(), "unit"] = ""
 
     # verify that there are no nan's left (in columns)
-    null_rows = df.isnull().values
+    null_rows = df.isnull().T.any()
     if null_rows.any():
-        _raise_data_error("empty cells in `data`", df.loc[null_rows])
+        _raise_data_error("Empty cells in `data`", df.loc[null_rows])
+    del null_rows
 
-    # check for duplicates and empty data
+    # format the time-column
+    df = format_time_col(df, time_col)
+
+    # cast to pd.Series, check for duplicates
     idx_cols = index + REQUIRED_COLS + [time_col] + extra_cols
-    rows = df[idx_cols].duplicated()
+    df = df.set_index(idx_cols).value
+
+    rows = df.index.duplicated()
     if any(rows):
-        _raise_data_error("duplicate rows in `data`", df.loc[rows, idx_cols])
+        _raise_data_error(
+            "Duplicate rows in `data`", df[rows].index.to_frame(index=False)
+        )
 
```

Comment:
In analogy to `del null_row` we could also delete rows, saving a little bit more memory
```suggestion
    del rows
```